freeipa/src/main/java/com/sequenceiq/freeipa/flow/chain/UpgradeFlowEventChainFactory.java

```diff
@@ -72,14 +72,15 @@ private List<Selectable> createScaleEventsAndChangePgw(UpgradeEvent event, int i
             Set<String> groupNames) {
         LOGGER.debug("Add events for primary gateway with id: [{}]", event.getPrimareGwInstanceId());
         List<Selectable> events = new ArrayList<>(PRIMARY_GW_EVENT_COUNT);
+        List<String> instanceIdToDownscale = Lists.newArrayList(event.getPrimareGwInstanceId());
         addMigrationFlowIfNeed(event, events, groupNames, "Resource create flow");
-        events.add(new UpscaleEvent(UpscaleFlowEvent.UPSCALE_EVENT.event(),
-                event.getResourceId(), instanceCountForUpscale, Boolean.FALSE, true, false, event.getOperationId(), event.getTriggeredVariant()));
-        ArrayList<String> oldInstances = new ArrayList<>(event.getInstanceIds());
+        events.add(new UpscaleEvent(UpscaleFlowEvent.UPSCALE_EVENT.event(), event.getResourceId(), instanceIdToDownscale,
+                instanceCountForUpscale, Boolean.FALSE, true, false, event.getOperationId(), event.getTriggeredVariant()));
+        List<String> oldInstances = new ArrayList<>(event.getInstanceIds());
```

Comment:
please don't change this to list to ensure events contain `ArrayList`, as it is serializable, while some kind of lists are not. 
see: https://github.com/hortonworks/cloudbreak/commit/4f4b62c69111dfd9b1bb4f68c17426e02244fa49